### ISSUE TITLE
[bug] StripeClient subscriptions & autoPagingIterator - missing parameters

### ISSUE THREAD

**Issue by armetiz** [stripe/stripe-php#939]
Hi there.

I'm migrating from 6.41 to 7.36.
Problem occurred when using autoPagingIterator on a subscriptions collection coming from a StripeClient. I didn't try on other types.

## Usages

```
// legacy code with singleton
$subscriptions = Stripe\Subscription::all([
    'status' => 'all',
    'limit' => 100,
]);

/** @var Stripe\Subscription $subscription */
foreach ($subscriptions->autoPagingIterator() as $subscription) {
    // subscription processing
}
```

```
// StripeClient version
$subscriptions = $this->client->subscriptions->all([
    'status' => 'all',
    'limit' => 100,
]);

/** @var Stripe\Subscription $subscription */
foreach ($subscriptions->autoPagingIterator() as $subscription) {
    // subscription processing
}
```

## Problem

The last subscription status from first iteration is `cancelled`.
When using StripeClient version, I get the following error : "No such subscription: sub_XXXX"

## Explanation

StripeClient autoPagingIterator does not forward `$param` from the initial request.

See `$param` argument values from the `ApiRequestor::request()` function.

### Legacy version

```
// first iteration
array:2 [
    "status" => "all"
    "limit" => 100
]

// second iteration
array:3 [
    "status" => "all"
    "limit" => 100
    "starting_after" => "sub_XXXX"
]
```

### StripeClient version

```
// first iteration
array:2 [
    "status" => "all"
    "limit" => 100
]

// second iteration
// note that status & limit are missing.
array:1 [
    "starting_after" => "sub_XXXX"
]
```

Edit : Explanation legacy / StripeClient have been inverted. Problem is still the same.

**Comment by remi-stripe** [comment]
@armetiz Thanks a lot for the detailed report, we're looking into it!

**Comment by armetiz** [comment]
I've just edited the explanation. Explanation have been inverted between legacy & StripeClient parameter values.

**Comment by richardm-stripe** [comment]
Should be fixed as of [v7.36.1](https://github.com/stripe/stripe-php/releases/tag/v7.36.1). Again, thank you @armetiz for the great write-up. It was very straightforward to understand and debug the issue.

**Comment by armetiz** [comment]
Welcome @richardm-stripe ;)
